### PR TITLE
bmips: add support for Actiontec R1000H

### PR DESCRIPTION
**Pull request by danitool**
The  Actiontec R1000H is a gigabit wifi router, 2.4 GHz single band with two external antennas. It comes with a coaxial HomePNA port.

Hardware:
 - SoC: Broadcom BCM6368
 - CPU: dual core BMIPS4350 V3.1 @400Mhz
 - RAM: 64 MB DDR
 - Flash: 32 MB parallel NOR
 - LAN switch: Broadcom BCM53115, 5x 1Gbit
 - LAN coaxial : 1x HPNA 3.1, CG3211 + CG3213
 - Wifi 2.4 GHz: Broadcom BCM4322 802.11bgn
 - USB: 1x 2.0
 - Buttons: 2x, 1 reset
 - LEDs: 7x
 - UART: yes

The HPNA hardware probably needs a firmware to make the coaxial port work. In the OEM firmware is apparently sent with an utility (inhpna) through the ethernet port.

Installation via CFE web UI:
  1. Connect the UART serial port.
  2. Power on the router and press enter at the console prompt to stop the bootloader.
  4. Browse to http://192.168.1.1 and upload the OpenWrt CFE firmware
  5. Wait a few minutes for it to finish

In bcm63xx this router didn't have support for the external gigabit switch. In bmpis it will be configurable allowing to route traffic on the wan port.

CC: @Noltari 
CC: @protectivedad

### CODE REVIEW

File: target/linux/bmips/dts/bcm6368-actiontec-r1000h.dts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+			port@5 {
+				/* HPNA coaxial port */
+				reg = <5>;
+				label = "coax";
+
+				phy-mode = "mii";
+
+				fixed-link {
+					speed = <100>;
```
Comment (Noltari) [inline target/linux/bmips/dts/bcm6368-actiontec-r1000h.dts:131]: Only 100 Mbps? :man_facepalming:

File: target/linux/bmips/image/bcm6368.mk [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+  DEVICE_VENDOR := Actiontec
+  DEVICE_MODEL := R1000H
+  CFE_BOARD_ID := 96368VVW
+  CHIP_ID := 6368
+  IMAGES += sysupgrade.bin
+  BLOCKSIZE := 0x20000
+  FLASH_MB := 32
+  CFE_EXTRAS += --signature "$$(DEVICE_VENDOR)"
+  DEVICE_PACKAGES += $(USB2_PACKAGES) \
+    $(B43_PACKAGES) \
```
Comment (Noltari) [inline target/linux/bmips/image/bcm6368.mk:14]: No SPROM needed? :astonished:

File: target/linux/bmips/dts/bcm6368-actiontec-r1000h.dts (hunk 1 above)
Comment (danitool) [reply target/linux/bmips/dts/bcm6368-actiontec-r1000h.dts:131]: Well, most likely 200 Mbps , but there is no kernel property for turbo MII. I guess this is configured in hardware at least in the BCM53115 switch if it was the case.

File: target/linux/bmips/image/bcm6368.mk (hunk 2 above)
Comment (danitool) [reply target/linux/bmips/image/bcm6368.mk:14]: It's a detachable miniPCI card. The SPROM wasn't needed in bcm63xx for this board.

**Review by Noltari — APPROVED** [review]
LGTM

File: target/linux/bmips/bcm6368/base-files/etc/board.d/02_network [hunk 3]
```diff
@@ -5,6 +5,10 @@
 board_config_update
 
 case "$(board_name)" in
+actiontec,r1000h)
+	ucidef_set_bridge_device switch
+	ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4 coax" "wan"
```
Comment (Noltari) [inline target/linux/bmips/bcm6368/base-files/etc/board.d/02_network:10]: Does it make sense to add `coax` here?
This should probably be added to a separate interface or not added at all if we can't get it working on OpenWrt...

Comment (Noltari) [reply target/linux/bmips/bcm6368/base-files/etc/board.d/02_network:10]: Something like https://github.com/openwrt/openwrt/blob/main/target/linux/bmips/bcm63268/base-files/etc/board.d/02_network#L16-L17

Comment (danitool) [reply target/linux/bmips/bcm6368/base-files/etc/board.d/02_network:10]: ok, I''ll make this change

File: target/linux/bmips/bcm6368/base-files/etc/board.d/01_leds [hunk 4]
```diff
@@ -6,6 +6,9 @@
 board_config_update
 
 case "$(board_name)" in
+actiontec,r1000h)
+	ucidef_set_led_usbport "usb" "USB" "green:usb" "usb1-port1" "usb2-port1"
+	ucidef_set_led_netdev "wan" "WAN" "green:wan" "wan"
```
Comment (Noltari) [inline target/linux/bmips/bcm6368/base-files/etc/board.d/01_leds:11]: I've just noticed that this is missing `;;`, but don't worry, I'll fix it myself :)

File: target/linux/bmips/image/bcm6368.mk [hunk 5]
```diff
@@ -1,5 +1,21 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+define Device/actiontec_r1000h
+  $(Device/bcm63xx-cfe)
+  DEVICE_VENDOR := Actiontec
+  DEVICE_MODEL := R1000H
+  CFE_BOARD_ID := 96368VVW
+  CHIP_ID := 6368
+  IMAGES += sysupgrade.bin
```
Comment (Noltari) [inline target/linux/bmips/image/bcm6368.mk:9]: And this line isn't needed.

**Pull request merged**